README.md

```diff
@@ -92,6 +92,39 @@ There are several other implementations available depending on your needs.
 | cce_kahan_full_c_full_e (cce_exact) | This additionally removes gradient filtering from the embedding gradient. This is useful as a reference point/sanity check. |
 
 
+### Vocabulary Parallelism
+
+We also support computing linear cross-entropy loss for classifier weights sharded
+along the vocabulary dimensions. To use this, simply provided a `VocabParallelOptions` instance
+to `linear_cross_entropy`. This takes 3 parameters, the `start` and `stop` indices of this ranks
+shard, and the `torch.distributed.ProcessGroup` that defines the vocab parallel group for this rank.
+
+
+
+```python
+import torch
+
+from cut_cross_entropy import linear_cross_entropy, VocabParallelOptions
+
+vp_group = torch.distributed.new_group(...)
+embeddings = model.compute_embedding(inputs)
+vp_classifier = model.get_classifier_weights()
+vp_start, vp_stop = model.get_classifier_range()
```

Comment:
Hmm this is maybe a bit tedious for the user to have to do, could this maybe be done automatically? If we have the `vp_group` we should know the following:
- total world size (we should always retrieve this info)
- vocab_size
- rank of each process
- group_size

=> Then we could retrieve the start and end automatically no?

chunk_size = world_size = len(group_size)
start_idx, end_indx = chunk_size * rank, chunk_size * (rank + 1)

Reply:
Ah ups sorry just saw your comment below? 😅 Ok perf making `vp_start` and `vp_stop` optional sounds great.